zerver/tests/test_auth_backends.py

```diff
@@ -5090,17 +5102,19 @@ def test_ldap_auth_email_auth_organization_restriction(self) -> None:
         # We do test two combinations here:
         # The first user has no (department) attribute set
         # The second user has one set, but to a different value
-        result = self.client_post(
-            "/api/v1/fetch_api_key",
-            dict(username="hamlet", password=self.ldap_password("hamlet")),
-        )
+        with self.artificial_transaction_savepoint():
```

Comment:
sprinkling this context manager in throughout ldap tests feels very ugly, but I'm not sure what else to do to deal with the `TransactionManagementError` issue that `atomic(savepoint=False)` causes in tests.

Reply:
Yeah, it's a bit ugly, but I think it's maybe OK -- the failure mode when you don't use it is at least very recognizable and we think only impacts tests. I think we can live with this.